tests/integration/no_unsafe_paths/BUILD.bazel

```diff
@@ -0,0 +1,34 @@
+# Copyright 2024 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load("@python//3.10:defs.bzl", py_test_3_10 = "py_test")
+load("@python//3.11:defs.bzl", py_test_3_11 = "py_test")
+
+py_test_3_10(
```

Comment:
Instead of an integration test, use `sh_py_run_test`. These are much lighter weight (don't require a bazel-within-bazel to run) and easier to run and debug.

See tests/bootstrap_impls/BUILD.bazel for examples. Do something akin to this:

```
load("//tests/support:sh_py_run_test.bzl", "py_reconfig_test")

py_reconfig_test(
  srcs=["whatever.py"],
  python_version="3.10",
  bootstrap_impl = "script",
  target_compatible_with = _SUPPORTS_BOOTSTRAP_SCRIPT
)
```

You can probably copy/paste this build file and replace the `py_test_XX` calls with `py_reconfig_test` and it should just work.

Adding them to `tests/bootstrap_impls` is probably easiest. A new directory, `tests/no_unsafe_paths` is also SGTM.

Reply:
These rules are neat - thanks!